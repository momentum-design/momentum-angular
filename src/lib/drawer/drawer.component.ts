 /** @component drawer */
import { Component, Input, Output, EventEmitter, HostBinding, ElementRef } from '@angular/core';

export type DrawerOpenFromDirection = 'right' | 'top' | 'bottom';
export type DrawerSize = 'medium' | 'large';

@Component({
  selector: 'md-drawer',
  template: `
    <ng-content></ng-content>
    <md-icon
      className="md-drawer__close"
      name="cancel_14"
      ariaLabel="{{closeAriaLabel}}"
      (click)="close()" >
    </md-icon>
  `,
  styles: []
})
export class DrawerComponent {

  private _visible: Boolean = false;

  /** @prop css class names | ''  */
  @Input() public class: string = '';

  @HostBinding('class') get className(): string {
    return 'md-drawer '
      + `${'md-drawer--' + this.openFrom} `
      + `${'md-drawer--' + this.openFrom + '--' + this.size } `
      + `${!this.isOpen ? 'md-drawer--' + this.openFrom + '--closed' : ''}`
      + `${this.class ? this.class : '' } `;
  }

   /** @prop Set whether the drawer is open or not | 'false' */
  @Input()
  set isOpen (value: Boolean) {
    this._visible = value;

  }
  get isOpen() {
    return this._visible;
  }

   /** @prop Sets the openFrom opens | 'right' */
  @Input() openFrom: DrawerOpenFromDirection = 'right';

   /** @prop Sets the openFrom opens defaults | 'medium */
   @Input() size: DrawerSize = 'medium';

   /** @prop set the text for accessibilty on the close button | 'close' */
   @Input() closeAriaLabel: String = 'close';

  /** use this event to know when the drawer has changed */
  @Output() drawerChanged: EventEmitter<Boolean> = new EventEmitter();

  close() {
    this.isOpen = false;
    this.drawerChanged.emit(this.isOpen);
  }

  constructor(el: ElementRef) {}

}
