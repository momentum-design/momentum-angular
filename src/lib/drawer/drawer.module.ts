import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DrawerComponent } from './drawer.component';
import { IconModule } from '../icon/icon.module';

@NgModule({
  declarations: [DrawerComponent],
  imports: [
    CommonModule,
    IconModule
  ],
  exports: [DrawerComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class DrawerModule { }
