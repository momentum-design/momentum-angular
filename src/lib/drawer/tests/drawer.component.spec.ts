import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { Component, DebugElement, ViewChild, ElementRef } from '@angular/core';
import { DrawerModule } from '../drawer.module';

describe('DrawerComponent', () => {
  let component: TestDrawerComponent;
  let fixture: ComponentFixture<TestDrawerComponent>;
  let drawerElement: HTMLElement;

  @Component({
    selector: 'test-drawer',
    template: `
      <button #triggerDrawer md-button aria-label="openDrawer" (click)="showMe=true;">
        Open Drawer
      </button>
      <md-drawer [(isOpen)]="showMe" (drawerChanged)="changeDrawer($event)"  >
        <p>Howdy</p>
      </md-drawer>`
  })
  class TestDrawerComponent {
    @ViewChild('triggerDrawer') triggerDrawer: ElementRef<HTMLElement>;
  }

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ TestDrawerComponent ],
      imports: [DrawerModule]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestDrawerComponent);
    component = fixture.componentInstance;
    const element = fixture.nativeElement;
    drawerElement = element.querySelector('md-drawer');
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(drawerElement).toBeTruthy();
    expect(drawerElement).toMatchSnapshot();
  });

  it ('should open when triggered', () => {
    const triggerDrawer = fixture.componentInstance.triggerDrawer.nativeElement;
    triggerDrawer.click();
    fixture.detectChanges();
    expect(drawerElement.className).toContain('md-drawer md-drawer--right md-drawer--right--medium ');
  });

});
