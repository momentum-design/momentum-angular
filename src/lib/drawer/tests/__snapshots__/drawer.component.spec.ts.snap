// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerComponent should create and match snapshot 1`] = `
<md-drawer
  class="md-drawer md-drawer--right md-drawer--right--medium md-drawer--right--closed "
>
  <p>
    Howdy
  </p>
  <md-icon
    aria-labelledby=""
    class=""
    classname="md-drawer__close"
    name="cancel_14"
    ng-reflect-aria-label="close"
    ng-reflect-class-name="md-drawer__close"
    ng-reflect-name="cancel_14"
    style="display:inline-flex;"
  >
    
    <button
      active="false"
      aria-label="close"
      class="md-button md-button--36 md-button--icon"
      href=""
      md-button=""
      ng-reflect-ng-class="md-button--icon"
      tabindex="0"
      type="button"
    >
      
      <span
        class="md-button__children"
        ng-reflect-ng-style="[object Object]"
        style="opacity: 1;"
      >
        <i
          class="md-icon icon icon-cancel_14 md-drawer__close"
        />
      </span>
    </button>
  </md-icon>
</md-drawer>
`;
